src/containers/BrowseMoviesContainer.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import BrowseMoviesPage from "../components/BrowseMoviesPage";
+import { getMoviesFromType, getGenreMovies } from "../api/APIUtils";
+
+class BrowseMoviesContainer extends Component {
+  state = { movies: [] };
+
+  componentDidMount() {
+    this.getMoviesFromTab();
+  }
+
+  componentDidUpdate() {
+    this.getMoviesFromTab();
+  }
+
+  getMoviesFromTab() {
+    if (this.props.match.params.filter === "top") {
```

Comment:
Add match to propTypes. Will also be easier to read if you do
```js
const { filter } = this.props.match.params;
```
and then use the `filter` variable in each `if`